src/AccountingModule.sol

```diff
@@ -4,59 +4,104 @@ pragma solidity ^0.8.28;
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { IAccountingToken } from "./AccountingToken.sol";
-import { IFlexStrategy } from "./FlexStrategy.sol";
 import { IVault } from "@yieldnest-vault/interface/IVault.sol";
 import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import { AccessControlUpgradeable } from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
 
 interface IAccountingModule {
-    event LowerBoundUpdated(uint16 newValue, uint16 oldValue);
-    event TargetApyUpdated(uint16 newValue, uint16 oldValue);
+    struct StrategySnapshot {
+        uint64 timestamp;
```

Comment:
just make it a `uint256`, you're not saving any slots by making it a `uint64` (and the evm internally uses `uint256`, so this `uint64` might make it more costly than a regular `uint256`).

Also, doing this can then allow doing this: https://github.com/yieldnest/yieldnest-flex-strategy/pull/4/files#r2163658299

so changing it here ot uint256 allows this: